test-clients/src/main/java/com/hedera/services/bdd/spec/HapiApiSpec.java

```diff
@@ -197,14 +197,36 @@ public void run() {
 	}
 
 	public boolean tryReinitializingFees() {
-		try {
-			ratesProvider.init();
-			feeCalculator.init();
-		} catch (Throwable t) {
-			log.error("Fees failed to initialize!", t);
-			return false;
+		int secsWait = 0;
+		if(ciPropsSource != null) {
+			String secs = setup().ciPropertiesMap().get("secondsWaitingServerUp");
+			if(secs != null) {
+				secsWait = Integer.parseInt(secs);
+				secsWait = secsWait >= 0 ? secsWait : 0;
+			}
 		}
-		return true;
+		while (secsWait >= 0 )  {
+			try {
+				ratesProvider.init();
+				feeCalculator.init();
+				return true;
```

Comment:
Shouldn't `true` be returned after the while loop?

Reply:
If the init of rates and fee calculation is successful, it should already return `true` before reaching the end of the while loop. Otherwise, the init is not successful.